### PR TITLE
[firebase_auth_web] Add null check to _fromJsUser.

### PR DESCRIPTION
**Pull request by ditman**
## Description

Prevent `null` users (unauthenticated) from breaking the `onAuthStateChanged` stream.

## Related Issues

* https://github.com/FirebaseExtended/flutterfire/issues/1685

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] If the pull request affects only one plugin, the PR title starts with the name of the plugin in brackets (e.g. [cloud_firestore])
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/FirebaseExtended/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Comment by ditman** [comment]
> LGTM, note that you'll need a CHANGELOG and pubspec.yaml update

I just realized I do, thanks for reminding me. There has also been some internal talk about using `jsify`, so I'm going to take some time and rewrite the test with package:js instead.

File: packages/firebase_auth/firebase_auth_web/pubspec.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 name: firebase_auth_web
 description: The web implementation of firebase_auth
 homepage: https://github.com/FirebaseExtended/flutterfire/tree/master/packages/firebase_auth/firebase_auth_web
-version: 0.1.1+3
+version: 0.1.2
```
Comment (collinjackson) [inline packages/firebase_auth/firebase_auth_web/pubspec.yaml]: Since you didn't add any new API surface area I think this should be 0.1.1+1

Comment (ditman) [reply packages/firebase_auth/firebase_auth_web/pubspec.yaml]: Yep, can do!

**Comment by ditman** [comment]
I'm going to merge this to unblock users; @hterkelsen I can revisit this with any changes you might have. Do hit me up.

**Pull request merged**